File: src/main/java/com/brandlive/qa/automation/events3/pages/admin/socialprofiledetailssetting/Directory.java
package com.brandlive.qa.automation.events3.pages.admin.socialprofiledetailssetting;

import com.brandlive.qa.automation.utils.Base;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

public class Directory extends Base {

    private void clickOnDirectoryCard() {
        String directory = "(//*[name()='svg' and @id='icon-arrow-right'])[1]";
        WebElement directoryCard = waitForDisplayedAndEnabled(By.xpath(directory), "Directory Card");
        directoryCard.click();
        log.info("Clicked on the Directory Card.");
    }

    public void toggleActionOnDirectoryAvatar(String toggleAction) {
        clickOnDirectoryCard();
        String toggleDirectoryAvatarON = "//div[@class='editable-nav-name']//following::div[@class='form-control switch on']";
        String toggleDirectoryAvatarOFF = "(//div[@class='editable-nav-name']//following::div[@class='form-control switch'])[2]";
        if (toggleAction.equals("ON")) {
            if (elementPresent(By.xpath(toggleDirectoryAvatarON))) {
                log.info("Already the Directory Avatar toggle is ON");
            } else {
                WebElement directoryAvatarToggleOff = waitForDisplayedAndEnabled(By.xpath(toggleDirectoryAvatarOFF), "Avatar Toggle OFF");
                directoryAvatarToggleOff.click();
                log.info("Enabled the Directory Avatar Toggle.");
            }
        } else {
            if (elementPresent(By.xpath(toggleDirectoryAvatarOFF))) {
                log.info("Already the Profile toggle is OFF.");
            } else {
                WebElement directoryAvatarToggleON = waitForDisplayedAndEnabled(By.xpath(toggleDirectoryAvatarON), "Profile toggle ON");
                directoryAvatarToggleON.click();
                log.info("Disabled the Directory Avatar toggle.");
            }
        }
    }
}

File: test/page-objects/streams/UI/directory.ts
import Page from "../../page"
import reporter from "../../../utils/reporter"

class Directory extends Page {

    constructor() {
        super()
    }

    // Directory toggle details
    get directoryCard() { return $(`(//*[name()='svg' and @id='icon-arrow-right'])[1]`) }
    get directoryToggleON() { return $(`(//div[text()='Enable Directory']//following::div[@class='form-control switch on'])[1]`) }
    get directoryToggleOFF() { return $(`(//div[text()='Enable Directory']//following::div[@class='form-control switch'])[1]`) }
    get avatarToggleON() { return $(`(//div[text()='Avatar']//following::div[@class='form-control switch on'])[1])`) }
    get avatarToggleOFF() { return $(`(//div[text()='Avatar']//following::div[@class='form-control switch'])[1]`) }
    get bookmarkAttendeesToggleON() { return $(`(//div[text()='Avatar']//following::div[@class='form-control switch on'])[1])`) }
    get bookmarkAttendeesToggleOFF() { return $(`(//div[text()='Avatar']//following::div[@class='form-control switch'])[1]`) }
    get filterToggleOFF() { return $(`(//div[text()='Filters']//following::div[@class='form-control switch'])[1]`) }
    get filterToggleON() { return $(`(//div[text()='Filters']//following::div[@class='form-control switch on'])[1]`) }
    get filterQuestion() { return $(`//div[@class='filter-tag not-selected tag-container']`) }
    get saveFilter() { return $(`//button[@class='button primary']`) }


    async clickOnDirectoryCard() {
        try {
            if (await this.directoryCard.isDisplayed()) {
                await this.click(await this.directoryCard)
                reporter.addStep("info", 'Clicked on the directory Card.')
            }
        } catch (err) {
            err.message = `Error clicking on directory card: , ${err.message} `
            throw err
        }
    }

    async directoryToggleAction(toggle: string) {
        try {
            if (toggle.match("ON")) {
                await this.click(await this.directoryToggleOFF)
                reporter.addStep("info", 'Enabled the directory toggle.')
            } else {
                await this.click(await this.directoryToggleON)
                reporter.addStep("info", 'Disabled the directory toggle.')
            }
        } catch (err) {
            err.message = `Error performing ${toggle} action on the directory toggle: , ${err.message} `
            throw err
        }
    }

    async avatarToggleAction(toggle: string) {
        try {
            if (toggle.match("ON")) {
                await this.click(await this.avatarToggleOFF)
                reporter.addStep("info", 'Enabled the avatar toggle.')
            } else {
                await this.click(await this.avatarToggleON)
                reporter.addStep("info", 'Disabled the avatar toggle.')
            }
        } catch (err) {
            err.message = `Error performing ${toggle} action on the avatar toggle: , ${err.message} `
            throw err
        }
    }

    async bookmarkAttendeesToggleAction(toggle: string) {
        try {
            if (toggle.match("ON")) {
                await this.click(await this.bookmarkAttendeesToggleOFF)
                reporter.addStep("info", 'Enabled the Bookmark Attendees toggle.')
            } else {
                await this.click(await this.bookmarkAttendeesToggleON)
                reporter.addStep("info", 'Disabled the directory toggle.')
            }
        } catch (err) {
            err.message = `Error performing ${toggle} action on the directory toggle: , ${err.message} `
            throw err
        }
    }

    /*
    * Select the added question as a filter
    */
    async selectFilterQuestion() {
        try {
            await this.click(await this.filterQuestion)
            reporter.addStep("info", "Selected the Filter Question.")
        } catch (err) {
            err.message = `Error selecting the filter question: , ${err.message} `
            throw err
        }
    }

    /*
    * Click On Save Filter button
    */
    async clickOnSaveFilter() {
        try {
            if (await this.saveFilter.isDisplayed()) {
                await this.click(await this.saveFilter)
                reporter.addStep("info", "Clicked on the Save Filter button.")
            }
        } catch (err) {
            err.message = `Error clicking on the Save Filter button.`
            throw err
        }
    }

    /*
    * Toggle action on Directory Filter
    */
    async filterToggleAction(toggle: string) {
        try {
            if (toggle.match("ON")) {
                await this.click(await this.filterToggleOFF)
                reporter.addStep("info", 'Enabled the Filter toggle.')
            } else {
                await this.click(await this.filterToggleON)
                reporter.addStep("info", 'Disabled the Filter toggle.')
            }
        } catch (err) {
            err.message = `Error performing ${toggle} action on the Filter toggle: , ${err.message} `
            throw err
        }
    }
}
export default new Directory